tests/lax_test.py

```diff
@@ -1161,6 +1161,58 @@ def testDotGeneralAgainstNumpy(self, lhs_shape, rhs_shape, dtype,
     numpy_op = lambda x, y: lax_reference.dot_general(x, y, dimension_numbers)
     self._CheckAgainstNumpy(numpy_op, op, args_maker)
 
+  def _testRaggedDot(self, m, k, n, num_groups, dtype, test_cb):
+    """Tests ragged_dot.
+
+    Creates input data and calls test_cb that runs the ragged_dot on the inputs and checks expectations.
+
+    Raises:
+      SkipTest: in the case dtype is not supported.
+    """
+    if np.issubdtype(dtype, np.bool_):
+      # This is due to "reduce_sum does not accept dtype bool".
+      raise SkipTest('ragged_dot for bool not supported')
+    rng = jtu.rand_small(self.rng())
+    lhs_shape = (m, k)
+    rhs_shape = (num_groups, k, n)
+    def group_sizes(m, num_groups):
+      ends_no_final = jnp.sort(np.random.choice(m, size=num_groups - 1))
```

Comment:
Should the random choice here use `rng`, or otherwise should the numpy RNG being used be seeded first?

Reply:
Thank you, switched to using `self.rng().choice()` instead.